internal/pkg/client/cache/library.go

```diff
@@ -6,13 +6,15 @@
 package cache
 
 import (
-	"fmt"
+	"errors"
 	"os"
 	"path/filepath"
 
 	"github.com/sylabs/scs-library-client/client"
 )
 
+var ErrCacheHashNotTheSame = errors.New("hash does not match")
```

Comment:
Maybe we should put this in `dir.go` since this error will now be part of the package API. Might also be worth calling it something like `ErrBadChecksum` so it's a little less verbose.